<launch>
  
    <!-- 送信先のVertual Motion TrackerのPCのIPとポート(ポートは標準が39570) -->
    <param name="vmt_ip" value="192.168.7.106" />
    <param name="vmt_port" value="39570" />

    <!-- URDFのパスのセット -->
    <param name="robot_description" textfile="$(find damitora_description)/urdf/damitora.urdf" />

    <!-- joint_state_publisher_gui -->
    <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" />

    <!-- だみとらOSC(トラッカー5個)の起動 -->
    <node name="damitora_osc" pkg="damitora_osc" type="damitora_osc" />

    <!-- robot_state_publisher -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

    <!-- Rviz 起動 -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find damitora_description)/rviz/damitora.rviz" />

    <!-- TFをposeに変換するノード(トラッカーの座標・姿勢はここで自動計算) -->
    <node name="pose_tracker_C" pkg="jsk_topic_tools" type="tf_to_pose.py">
        <param name="src_frame" value="world"/>
        <param name="dst_frame" value="tracker_C_link"/>
        <param name="rate" value="60"/>
    </node>

    <!-- TFをposeに変換するノード(トラッカーの座標・姿勢はここで自動計算) -->
    <node name="pose_tracker_R" pkg="jsk_topic_tools" type="tf_to_pose.py">
        <param name="src_frame" value="world"/>
        <param name="dst_frame" value="tracker_R_link"/>
        <param name="rate" value="60"/>
    </node>

    <!-- TFをposeに変換するノード(トラッカーの座標・姿勢はここで自動計算) -->
    <node name="pose_tracker_L" pkg="jsk_topic_tools" type="tf_to_pose.py">
        <param name="src_frame" value="world"/>
        <param name="dst_frame" value="tracker_L_link"/>
        <param name="rate" value="60"/>
    </node>

    <!-- TFをposeに変換するノード(トラッカーの座標・姿勢はここで自動計算) -->
    <node name="pose_tracker_R2" pkg="jsk_topic_tools" type="tf_to_pose.py">
        <param name="src_frame" value="world"/>
        <param name="dst_frame" value="tracker_R_link2"/>
        <param name="rate" value="60"/>
    </node>

    <!-- TFをposeに変換するノード(トラッカーの座標・姿勢はここで自動計算) -->
    <node name="pose_tracker_L2" pkg="jsk_topic_tools" type="tf_to_pose.py">
        <param name="src_frame" value="world"/>
        <param name="dst_frame" value="tracker_L_link2"/>
        <param name="rate" value="60"/>
    </node>

</launch>
